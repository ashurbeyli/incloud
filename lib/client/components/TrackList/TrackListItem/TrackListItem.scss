.tracklist-item__id {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  direction: rtl;
}
.tracklist-item__actions {
  button {
    margin: 0 5px;
    float: right;
  }
}