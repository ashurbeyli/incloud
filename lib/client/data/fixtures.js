export const dumbTask = {
  description: '',
  pieces: [],
};

export const task = {
  createdAt: '2018-04-19T22:07:34.639Z',
  description: 'asdfafa',
  pieces: [],
  updatedAt: '2018-04-19T22:07:34.640Z',
  user_id: '5ad65aa44212e29c4a922af7',
  __v: 0,
  _id: '5ad91326115be0385d527a1b'
};

export const tasks = [
  task,
  {
    createdAt: '2018-04-19T22:07:34.639Z',
    description: 'asdfafa',
    pieces: [],
    updatedAt: '2018-04-19T22:07:34.640Z',
    user_id: '5ad65aa44212e29c4a922af7',
    __v: 0,
    _id: '5ad916889a0c1f39b919eff7'
  },
];