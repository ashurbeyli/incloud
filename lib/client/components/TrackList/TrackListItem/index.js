import React from 'react';

import { Button, Col, Glyphicon, Grid, ListGroupItem, Row } from 'react-bootstrap';
import Moment from 'react-moment';
import { fullRecordedTime } from '../../../utils/time';

import './TrackListItem.scss';

export const TrackListItem = ({ task, onDelete }) => {

  return (
    <ListGroupItem key={task._id} className="tracklist-item">
      <Grid>
        <Row className="show-grid">
          <Col xs={12} md={1} className="tracklist-item__id">
            {task._id}
          </Col>
          <Col xs={12} md={5} className="tracklist-item__description">
            {task.description}
          </Col>
          <Col xs={12} md={2} className="tracklist-item__datetime">
            <Moment format="DD MMM YYYY hh:mm">
              {task.createdAt}
            </Moment>
          </Col>
          <Col xs={12} md={1} className="tracklist-item__datetime">
            {fullRecordedTime(task.pieces, null)}
          </Col>
          <Col xs={6} md={2} className="tracklist-item__actions">
            <Button
              bsSize="small"
              bsStyle="danger"
              onClick={() => onDelete(task._id)}
            ><Glyphicon glyph="trash" /></Button>
          </Col>
        </Row>
      </Grid>
    </ListGroupItem>
  );

};

export default TrackListItem;