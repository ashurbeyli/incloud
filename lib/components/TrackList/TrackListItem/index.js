import React from 'react';

import { Button, Col, Glyphicon, Grid, ListGroupItem, Row } from 'react-bootstrap';
import Moment from 'react-moment';

import './TrackListItem.scss';

export const TrackListItem = ({ task, onDelete }) => {

  return (
    <ListGroupItem key={task._id} className="tracklist-item">
      <Grid>
        <Row className="show-grid">
          <Col xs={12} md={2} className="tracklist-item__id">
            {task._id}
          </Col>
          <Col xs={12} md={6} className="tracklist-item__description">
            {task.description}
          </Col>
          <Col xs={12} md={2} className="tracklist-item__datetime">
            <Moment format="DD MMM YYYY hh:mm">
              {task.createdAt}
            </Moment>
          </Col>
          <Col xs={6} md={2} className="tracklist-item__actions">
            {/*<Button bsSize="small" bsStyle="primary"><Glyphicon glyph="play" /></Button>*/}
            <Button
              bsSize="small"
              bsStyle="danger"
              onClick={() => onDelete(task._id)}
            ><Glyphicon glyph="trash" /></Button>
          </Col>
        </Row>
      </Grid>
    </ListGroupItem>
  );

};

export default TrackListItem;