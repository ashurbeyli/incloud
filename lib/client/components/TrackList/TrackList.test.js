import React from 'react';

import { TrackList } from './';
import { tasks } from '../../data/fixtures';
import TrackListItem from './TrackListItem/index';

const props = { tasks };

describe('TrackList', () => {
  const wrapper = shallow(<TrackList {...props}/>);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`should have ${props.tasks.length} TrackListItems`, () => {
    expect(wrapper.find(TrackListItem).length).toEqual(props.tasks.length);
  });

});