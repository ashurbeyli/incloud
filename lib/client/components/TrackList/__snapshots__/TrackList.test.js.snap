// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrackList
    tasks={
      Array [
        Object {
          "__v": 0,
          "_id": "5ad91326115be0385d527a1b",
          "createdAt": "2018-04-19T22:07:34.639Z",
          "description": "asdfafa",
          "pieces": Array [],
          "updatedAt": "2018-04-19T22:07:34.640Z",
          "user_id": "5ad65aa44212e29c4a922af7",
        },
        Object {
          "__v": 0,
          "_id": "5ad916889a0c1f39b919eff7",
          "createdAt": "2018-04-19T22:07:34.639Z",
          "description": "asdfafa",
          "pieces": Array [],
          "updatedAt": "2018-04-19T22:07:34.640Z",
          "user_id": "5ad65aa44212e29c4a922af7",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "list-group",
      "children": Array [
        <Unknown
          onDelete={[Function]}
          task={
            Object {
              "__v": 0,
              "_id": "5ad91326115be0385d527a1b",
              "createdAt": "2018-04-19T22:07:34.639Z",
              "description": "asdfafa",
              "pieces": Array [],
              "updatedAt": "2018-04-19T22:07:34.640Z",
              "user_id": "5ad65aa44212e29c4a922af7",
            }
          }
        />,
        <Unknown
          onDelete={[Function]}
          task={
            Object {
              "__v": 0,
              "_id": "5ad916889a0c1f39b919eff7",
              "createdAt": "2018-04-19T22:07:34.639Z",
              "description": "asdfafa",
              "pieces": Array [],
              "updatedAt": "2018-04-19T22:07:34.640Z",
              "user_id": "5ad65aa44212e29c4a922af7",
            }
          }
        />,
      ],
      "className": "tracklist",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "5ad91326115be0385d527a1b",
        "nodeType": "function",
        "props": Object {
          "onDelete": [Function],
          "task": Object {
            "__v": 0,
            "_id": "5ad91326115be0385d527a1b",
            "createdAt": "2018-04-19T22:07:34.639Z",
            "description": "asdfafa",
            "pieces": Array [],
            "updatedAt": "2018-04-19T22:07:34.640Z",
            "user_id": "5ad65aa44212e29c4a922af7",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5ad916889a0c1f39b919eff7",
        "nodeType": "function",
        "props": Object {
          "onDelete": [Function],
          "task": Object {
            "__v": 0,
            "_id": "5ad916889a0c1f39b919eff7",
            "createdAt": "2018-04-19T22:07:34.639Z",
            "description": "asdfafa",
            "pieces": Array [],
            "updatedAt": "2018-04-19T22:07:34.640Z",
            "user_id": "5ad65aa44212e29c4a922af7",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "list-group",
        "children": Array [
          <Unknown
            onDelete={[Function]}
            task={
              Object {
                "__v": 0,
                "_id": "5ad91326115be0385d527a1b",
                "createdAt": "2018-04-19T22:07:34.639Z",
                "description": "asdfafa",
                "pieces": Array [],
                "updatedAt": "2018-04-19T22:07:34.640Z",
                "user_id": "5ad65aa44212e29c4a922af7",
              }
            }
          />,
          <Unknown
            onDelete={[Function]}
            task={
              Object {
                "__v": 0,
                "_id": "5ad916889a0c1f39b919eff7",
                "createdAt": "2018-04-19T22:07:34.639Z",
                "description": "asdfafa",
                "pieces": Array [],
                "updatedAt": "2018-04-19T22:07:34.640Z",
                "user_id": "5ad65aa44212e29c4a922af7",
              }
            }
          />,
        ],
        "className": "tracklist",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "5ad91326115be0385d527a1b",
          "nodeType": "function",
          "props": Object {
            "onDelete": [Function],
            "task": Object {
              "__v": 0,
              "_id": "5ad91326115be0385d527a1b",
              "createdAt": "2018-04-19T22:07:34.639Z",
              "description": "asdfafa",
              "pieces": Array [],
              "updatedAt": "2018-04-19T22:07:34.640Z",
              "user_id": "5ad65aa44212e29c4a922af7",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5ad916889a0c1f39b919eff7",
          "nodeType": "function",
          "props": Object {
            "onDelete": [Function],
            "task": Object {
              "__v": 0,
              "_id": "5ad916889a0c1f39b919eff7",
              "createdAt": "2018-04-19T22:07:34.639Z",
              "description": "asdfafa",
              "pieces": Array [],
              "updatedAt": "2018-04-19T22:07:34.640Z",
              "user_id": "5ad65aa44212e29c4a922af7",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
